account_mtd/models/mtd_hello_world.py

```diff
@@ -0,0 +1,406 @@
+
+        if api_token:
+            api_token.authorisation_code = auth_code
+        else:
+            api_token.create({
+                'api_id': api_tracker.api_id,
+                'api_name': api_tracker.api_name,
+                'authorisation_code': auth_code
+                })
+        # self.current_record = record_id
+        record = self.env['mtd.hello_world'].search([('id', '=', record_id)])
+        token_location_uri = "https://test-api.service.hmrc.gov.uk/oauth/token"
+        client_id = record.hmrc_configuration.client_id
+        client_secret = record.hmrc_configuration.client_secret
+        redirect_uri = "http://localhost:8090{}".format('/auth-redirect')
+
+        data_user_info = {
+            'grant_type': 'authorization_code',
+            'client_id': client_id,
+            'client_secret': client_secret,
+            'redirect_uri': redirect_uri,
+            'code': auth_code
+            }
+        self._logger.info(
+            "(Step 2) exchange authorisation code - Data which will be " +
+            "sent in the request:-  {}".format(json.dumps(data_user_info))
+        )
+        headers = {
+            'Content-Type': 'application/json',
+            }
+        self._logger.info(
+            "(Step 2) exchange authorisation code - headers which will be "
+            "sent in the request:-  {}".format(headers)
+        )
+        req = requests.post(token_location_uri, data=json.dumps(data_user_info), headers=headers)
+        response_token = json.loads(req.text)
+        self._logger.info(
+            "(Step 2) exchange authorisation code - received " 
+            "response of the request:- {}, and its text:- {}".format(req, response_token)
+        )
+        if req.ok:
+            # get the record which we created when sending the request and update the response_received column
+            # As this determines whether we can place another request or not
+            record_tracker = self.env['mtd.api_request_tracker'].search([
+                ('request_sent', '=', True),
+                ('response_received', '=', False)
+            ])
+            record_tracker.response_received = True
+            if not api_token:
+                api_token = self.env['mtd.api_tokens'].search([('authorisation_code', '=', auth_code)])
+            self._logger.info(
+                "(Step 2) exchange authorisation code - api_token table id where info is stored:- {}".format(api_token)
+            )
+            api_token.access_token = response_token['access_token']
+            api_token.refresh_token = response_token['refresh_token']
+            api_token.expires_in = json.dumps(response_token['expires_in'])
+            api_token.access_token_recieved_date = datetime.now()
+            version = self._json_command('version', record_id=record_id)
+            return version
+        else:
+            construct_text = (
```

Comment:
`error_message` a better name.
Consider one long formatting string with a single `format()` at the end and named params.